cli.ts

```diff
@@ -758,8 +783,10 @@ const preprocessArgs = (inputArgs: DenoArgs): DenoArgs => {
 	});
 };
 
-export const run = (env: EnvVars, inputArgs: DenoArgs, i18n: i18n.t) => {
+export const run = async (env: EnvVars, inputArgs: DenoArgs, i18n: i18n.t) => {
 	try {
+		await sendEvent(inputArgs, getVersion(inputArgs, i18n), inputArgs.includes('--fromVsCode') ? 'VSCode' : 'CLI');
```

Comment:
Is the event sending blocking?
Can we not await this code, so that:

1. The rest of code can run immediately without delay
2. Inn case of an error in sending the event, the main task would not fail

Reply:
It doesn't seem to fail even if there's a network error. I tried using it with my wifi off. I tried not having await but events won't be sent to GA for some reason. Maybe because the promise is not ever resolved? Idk exactly know why but it should resolve and send the events.

I am in the process of refactoring sendEvent to make the entire thing into a Future as per @mweichert suggestion.

Reply:
Turned into a Future and non-blocking